pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -613,8 +616,19 @@ public static void addColumnMetadataInfo(PropertiesConfiguration properties, Str
       properties.setProperty(getKeyFor(column, DATETIME_GRANULARITY), dateTimeFieldSpec.getGranularity());
     }
 
+    // complex field
+    if (fieldSpec.getFieldType().equals(FieldType.COMPLEX)) {
+      ComplexFieldSpec complexFieldSpec = (ComplexFieldSpec) fieldSpec;
+      properties.setProperty(getKeyFor(column, COMPLEX_CHILD_FIELD_NAMES),
+          new ArrayList<>(complexFieldSpec.getChildFieldSpecs().keySet()));
+      for (Map.Entry<String, FieldSpec> entry : complexFieldSpec.getChildFieldSpecs().entrySet()) {
+        addFieldSpec(properties, ComplexFieldSpec.getFullChildName(column, entry.getKey()), entry.getValue());
+      }
+    }
+
+    addFieldSpec(properties, column, fieldSpec);
```

Comment:
I'm a little bit confused. Are we adding the same properties again?

Reply:
removed.